lib/active_admin/filters/humanized.rb

```diff
@@ -9,8 +9,26 @@ def initialize(param)
         @value = param[1]
       end
 
+      def related_class
+        @related_class ||=
+            begin
```

Comment:
@Fivell Just to clarify, are you suggesting we move:

``` ruby
@body.split('_id')...
```
up to `ActiveAdmin::Filters::Active` and inject the corresponding classes into the filters array? Something like this:

``` ruby
      def build_filters
        filters = @params['q'] || []
        filters = filters.map { |f| f << deduce_resource_class(f) }
        filters.map { |param| Humanized.new(param) }
      end
```